lib/rules/key-spacing.js

```diff
@@ -47,6 +47,40 @@ function last(arr) {
     return arr[arr.length - 1];
 }
 
+/**
+ * Checks whether a property is a member of the property group it follows.
+ * @param {ASTNode} lastMember The last Property known to be in the group.
+ * @param {ASTNode} candidate The next Property that might be in the group.
+ * @returns {boolean} True if the candidate property is part of the group.
+ */
+function continuesPropertyGroup(lastMember, candidate) {
+    var groupEndLine = lastMember.loc.end.line,
+        candidateStartLine = candidate.loc.start.line,
+        comments, i;
+
+    if (candidateStartLine - groupEndLine <= 1) {
+        return true;
+    }
+
+    // Check that the first comment is adjacent to the end of the group, the
+    // last comment is adjacent to the candidate property, and that successive
+    // comments are adjacent to each other.
+    if (
+        (comments = candidate.leadingComments) &&
+        comments[0].loc.start.line - groupEndLine <= 1 &&
```

Comment:
Can you move the assignment outside the condition? The condition is complicated enough already.